### ISSUE TITLE
Rendering Pipeline Slowdowns on amdgpu / 0.41.x

### ISSUE THREAD

**Issue by pulsar256** [hyprwm/Hyprland#6485]
### Bug or Regression?

Bug

### System Info and Version

```
Hyprland, built from branch main at commit 5de273a14427cb4a4cad9ac57a22b418bcd4248d  (xwayland: drop some spammy logs to trace).
Date: Thu Jun 13 17:32:32 2024
Tag: v0.41.1-3-g5de273a1, commits: 4821

flags: (if any)


System Information:
System name: Linux
Node name: snowtronix
Release: 6.9.4-zen1-1-zen
Version: #1 ZEN SMP PREEMPT_DYNAMIC Wed, 12 Jun 2024 20:16:55 +0000


GPU information: 
26:00.0 VGA compatible controller [0300]: Advanced Micro Devices, Inc. [AMD/ATI] Ellesmere [Radeon RX 470/480/570/570X/580/580X/590] [1002:67df] (rev e7) (prog-if 00 [VGA controller])


os-release: NAME="Arch Linux"
PRETTY_NAME="Arch Linux"
ID=arch
BUILD_ID=rolling
ANSI_COLOR="38;2;23;147;209"
HOME_URL="https://archlinux.org/"
DOCUMENTATION_URL="https://wiki.archlinux.org/"
SUPPORT_URL="https://bbs.archlinux.org/"
BUG_REPORT_URL="https://gitlab.archlinux.org/groups/archlinux/-/issues"
PRIVACY_POLICY_URL="https://terms.archlinux.org/docs/privacy-policy/"
LOGO=archlinux-logo


plugins:

```

```
> uname -a
Linux snowtronix 6.9.4-zen1-1-zen #1 ZEN SMP PREEMPT_DYNAMIC Wed, 12 Jun 2024 20:16:55 +0000 x86_64 GNU/Linux
```

### Description

0.41.x release exhibits severe performance issues, presumably due to dmabuf initialization issues? Affects chromium based clients (electron apps, vivaldi) running in native wayland mode.

Bisected this issue down to:

```
722b846ac5205ec3e9a2940abfefa273fa4219d4 is the first bad commit
commit 722b846ac5205ec3e9a2940abfefa273fa4219d4 (HEAD)
Author: Vaxry <vaxry@vaxry.net>
Date:   Sun Jun 9 21:10:46 2024 +0200

    egl: assume implicit modifiers are available for old drivers

    fixes #6367

 src/Compositor.cpp               | 6 +++---
 src/managers/ProtocolManager.cpp | 9 +++++++--
 src/render/OpenGL.cpp            | 6 +++++-
 3 files changed, 15 insertions(+), 6 deletions(-)

snowtronix in development/priv/hyprland on  HEAD (722b846) 
```

when reverting the changes of this commit in `OpenGL.cpp` and `ProtocolManager.cpp` performance is back to normal and I can use dmabuf in OBS. With the changes in place, OBS crashes with

```
                Module libintlc.so.5 without build-id.
                Module libimf.so without build-id.
                Module libirng.so without build-id.
                Module libsvml.so without build-id.
                Module libOpenCL.so.1 without build-id.
                Stack trace of thread 41512:
                #0  0x0000767e64d2706d wl_proxy_marshal_constructor (libwayland-client.so.0 + 0x806d)
                #1  0x0000767e32a0616b n/a (wlrobs.so + 0x416b)
                #2  0x0000767e6c56e8ee n/a (libobs.so.0 + 0x738ee)
                #3  0x0000767e6c56efc0 obs_source_video_render (libobs.so.0 + 0x73fc0)
                #4  0x0000767e6c5626ee n/a (libobs.so.0 + 0x676ee)
                #5  0x0000767e6c56e8ee n/a (libobs.so.0 + 0x738ee)
                #6  0x0000767e6c56edad obs_source_video_render (libobs.so.0 + 0x73dad)
                #7  0x0000767e6c57b0b5 n/a (libobs.so.0 + 0x800b5)
                #8  0x0000767e3e009eee n/a (obs-transitions.so + 0x3eee)
                #9  0x0000767e6c56e8ee n/a (libobs.so.0 + 0x738ee)
                #10 0x0000767e6c56efc0 obs_source_video_render (libobs.so.0 + 0x73fc0)
                #11 0x0000767e6c58694d obs_view_render (libobs.so.0 + 0x8b94d)
                #12 0x0000767e6c57bfb9 obs_graphics_thread_loop (libobs.so.0 + 0x80fb9)
                #13 0x0000767e6c57d4a9 obs_graphics_thread (libobs.so.0 + 0x824a9)
                #14 0x0000767e674a6ded n/a (libc.so.6 + 0x92ded)
                #15 0x0000767e6752a0dc n/a (libc.so.6 + 0x1160dc)

                Stack trace of thread 41484:
                #0  0x0000767e6757b6ca n/a (libc.so.6 + 0x1676ca)
                #1  0x0000767e5479f757 _ZN15QtWaylandClient23QWaylandShmBackingStore6resizeERK5QSize (libQt6WaylandClient.so.6 + 0x8b757)
                #2  0x0000767e5479f845 _ZN15QtWaylandClient23QWaylandShmBackingStore10beginPaintERK7QRegion (libQt6WaylandClient.so.6 + 0x8b845)
                #3  0x0000767e681ff788 _ZN13QBackingStore10beginPaintERK7QRegion (libQt6Gui.so.6 + 0x1ff788)
                #4  0x0000767e68b5b880 _ZN21QWidgetRepaintManager13paintAndFlushEv (libQt6Widgets.so.6 + 0x15b880)
                #5  0x0000767e68b4b908 _ZN7QWidget5eventEP6QEvent (libQt6Widgets.so.6 + 0x14b908)
                #6  0x0000767e68afc55c _ZN19QApplicationPrivate13notify_helperEP7QObjectP6QEvent (libQt6Widgets.so.6 + 0xfc55c)
                #7  0x0000767e67b44e38 _ZN16QCoreApplication15notifyInternal2EP7QObjectP6QEvent (libQt6Core.so.6 + 0x144e38)
                #8  0x0000767e67b451fb _ZN23QCoreApplicationPrivate16sendPostedEventsEP7QObjectiP11QThreadData (libQt6Core.so.6 + 0x1451fb)
                #9  0x0000767e67da460c n/a (libQt6Core.so.6 + 0x3a460c)
                #10 0x0000767e6610ea89 n/a (libglib-2.0.so.0 + 0x5ca89)
                #11 0x0000767e661709b7 n/a (libglib-2.0.so.0 + 0xbe9b7)
                #12 0x0000767e6610df95 g_main_context_iteration (libglib-2.0.so.0 + 0x5bf95)
                #13 0x0000767e67da28dd _ZN20QEventDispatcherGlib13processEventsE6QFlagsIN10QEventLoop17ProcessEventsFlagEE (libQt6Core.so.6 + 0x3a28dd)
                #14 0x0000767e67b4f10e _ZN10QEventLoop4execE6QFlagsINS_17ProcessEventsFlagEE (libQt6Core.so.6 + 0x14f10e)
                #15 0x0000767e67b4945d _ZN16QCoreApplication4execEv (libQt6Core.so.6 + 0x14945d)
                #16 0x00005e892c2cee19 main (obs + 0x8be19)
                #17 0x0000767e67439c88 n/a (libc.so.6 + 0x25c88)
                #18 0x0000767e67439d4c __libc_start_main (libc.so.6 + 0x25d4c)
                #19 0x00005e892c2d2545 n/a (obs + 0x8f545)
```

reverting the changes on main improves the chromium based browsers performance again but breaks dmabuf in obs and alacritty. 

### How to reproduce

see description and logs

### 0.41.x

https://github.com/hyprwm/Hyprland/assets/1245126/c589462f-6ea4-4397-bcbb-acf392c72b96

[hyprland-5de273a1.log](https://github.com/user-attachments/files/15827551/hyprland-5de273a1.log)

### Demo 0.40.0
https://github.com/hyprwm/Hyprland/assets/1245126/c163f385-57b9-425b-ba2c-f27114144f94

[eglinfo.txt](https://github.com/user-attachments/files/15827583/eglinfo.txt)

**Comment by vaxerski** [comment]
please post a hyprland log pls uwu

**Comment by EvilBoi123** [comment]
yup agree i got the same issue. Apparently this issue occurs when you play gpu demanding game.

**Comment by vaxerski** [comment]
oh uh can one of you provide a hyprland log but with `HYPRLAND_LOG_WLR=1`?

**Comment by EvilBoi123** [comment]
OP already provided their own log. But here's mine with me playing game with `HYPRLAND_LOG_WLR=1`. 
[hyprland.log](https://github.com/user-attachments/files/15838304/hyprland.log)

 tldr from my log i can't figure out what causes it, idk if it be enough. is there any other request of logs method i can show of?

**Comment by vaxerski** [comment]
the log is truncated, please attach the entire log

**Comment by pulsar256** [comment]
@vaxerski 

capture of a session with `HYPRLAND_LOG_WLR=1` - opening chromium and obs to validate issue. (obs crashes on dmabuf capture config, chromium slowdown)

[hyprland-with-wlr-1.log](https://github.com/user-attachments/files/15839935/hyprland-with-wlr-1.log)

**Comment by vaxerski** [comment]
hm, nothing seems amiss. Can you launch chromium with WAYLAND_DEBUG=1 on hyprland, and on hyprland with the patch reverted, and send the two?

`WAYLAND_DEBUG=1 chromium --enable-features=UseOzonePlatform --ozone-platform=wayland |& tee ~/chromium.log`

**Comment by pulsar256** [comment]
> hm, nothing seems amiss. Can you launch chromium with WAYLAND_DEBUG=1 on hyprland, and on hyprland 0.40.0 and send the two?
> 
[chromium-0.40.0.log](https://github.com/user-attachments/files/15840872/chromium-0.40.0.log)
[chromium-0.41.x.log](https://github.com/user-attachments/files/15840873/chromium-0.41.x.log)

> `WAYLAND_DEBUG=1 chromium --enable-features=UseOzonePlatform --ozone-platform=wayland |& tee ~/chromium.log`

**Comment by vaxerski** [comment]
what?

**Comment by vaxerski** [comment]
[patch.txt](https://github.com/user-attachments/files/15842482/patch.txt)
try this

**Comment by pulsar256** [comment]
performance is the same

```
MESA: error: zink: could not create swapchain
```

when launching alacritty or using dmabuf in obs

**Comment by vaxerski** [comment]
[patch.txt](https://github.com/user-attachments/files/15842727/patch.txt)
this?

**Comment by pulsar256** [comment]
chromium issues solved, thanks! obs still crashing on dmabuf access but might be dealt with in a different issue as it is not strictly "bad performance".
